src/xisfreader.rs

```diff
@@ -77,92 +77,105 @@ fn xisf_parse_xml(
+                                let split = xisf_header.compression.split(':');
+                                for (n, s) in split.enumerate() {
+                                    match n {
+                                        0 => xisf_header.compression_codec = s.to_string(),
+                                        1 => xisf_header.compression_size = s.parse().unwrap(),
+                                        _ => (),
+                                    }
+                                }
+                            }
+                            _ => {} //name => eprintln!("unknown attribute name {}", name),
                         }
                     }
                 }
-            } else if node.tag_name().name() == "FITSKeyword" {
-                // Parse and store the values of the FITS keyword
-                let mut xisf_fits_keyword = FITSKeyword {
-                    name: String::from(""),
-                    value: String::from(""),
-                    comment: String::from(""),
-                };
-                for attr in node.attributes() {
-                    if attr.name() == "name" {
-                        xisf_fits_keyword.name = attr.value().to_string();
-                    } else if attr.name() == "value" {
-                        xisf_fits_keyword.value = attr.value().to_string();
-                    } else if attr.name() == "comment" {
-                        xisf_fits_keyword.comment = attr.value().to_string();
+                "FITSKeyword" => {
+                    // Parse and store the values of the FITS keyword
+                    let mut xisf_fits_keyword = FITSKeyword {
+                        name: String::from(""),
+                        value: String::from(""),
+                        comment: String::from(""),
+                    };
+                    for attr in node.attributes() {
+                        if attr.name() == "name" {
+                            xisf_fits_keyword.name = attr.value().to_string();
+                        } else if attr.name() == "value" {
+                            xisf_fits_keyword.value = attr.value().to_string();
+                        } else if attr.name() == "comment" {
+                            xisf_fits_keyword.comment = attr.value().to_string();
+                        }
                     }
+                    println!(
+                        "FITS Keyword: {} = {} / {}",
+                        xisf_fits_keyword.name, xisf_fits_keyword.value, xisf_fits_keyword.comment
+                    );
+                    xisf_fits_keywords.push(xisf_fits_keyword);
                 }
-                println!(
-                    "FITS Keyword: {} = {} / {}",
-                    xisf_fits_keyword.name, xisf_fits_keyword.value, xisf_fits_keyword.comment
-                );
-                xisf_fits_keywords.push(xisf_fits_keyword);
+                _ => {} // tag => eprintln!("unknown tag {}", tag);
```

Comment:
Something similar happens here:
```rust
_ => {}
```
That's the current one: don't do anything if the tag is not one of the previous ones, or:
```rust
tag => eprintln!("unknown tag {}", tag),
```
Which will print a message in `stderr` saying that the tag that was found is not known.

Reply:
I prefer the first option. There are other tags that aren't parsed right now, as they are not required.